data-model/src/main/java/io/micronaut/data/model/query/builder/sql/SqlQueryBuilder.java

```diff
@@ -483,9 +486,73 @@ public static boolean isForeignKeyWithJoinTable(@NonNull Association association
             }
         }
         createStatements.add(builder.toString());
+        addIndexes(entity, tableName, createStatements);
         return createStatements.toArray(new String[0]);
     }
 
+    private void addIndexes(PersistentEntity entity, String tableName, List<String> createStatements) {
+        final String indexes = createIndexesStatements(entity, tableName);
+        if (indexes.length() > 0) {
+            createStatements.add(indexes);
+        }
+    }
+
+    private String createIndexesStatements(PersistentEntity entity, String tableName) {
+
+        StringBuilder indexBuilder = new StringBuilder();
+
+       Optional<AnnotationValue<Index>[]> indexes = entity
+                .getAnnotationMetadata()
+                .findDeclaredAnnotation("javax.persistence.Table")
+                .flatMap(table -> Optional.ofNullable((AnnotationValue<Index>[]) table.getValues().get("indexes")));
+
+       Stream.of(indexes)
+               .flatMap(o -> o.map(Stream::of).orElseGet(Stream::empty))
+               .map(AnnotationValue::getValues)
+               .forEach(index -> addIndex(indexBuilder, index, tableName));
+
+       return indexBuilder.toString();
+
+    }
+
+    private void addIndex(StringBuilder indexBuilder, Map<CharSequence, Object> index, String tableName) {
```

Comment:
It's better to work with `AnnotationValue` instead of `Map<CharSequence, Object> ` that should eliminate:
` .ofNullable((Boolean) index.get("unique"))`